x-pack/plugins/lens/public/editor_frame_service/editor_frame/config_panel/config_panel.tsx

```diff
@@ -134,37 +134,42 @@ export function LayerPanels(
     [dispatch]
   );
 
+  const datasourcePublicAPIs = props.framePublicAPI.datasourceLayers;
+
   return (
     <EuiForm className="lnsConfigPanel">
-      {layerIds.map((layerId, index) => (
```

Comment:
Why is this change necessary?

Reply:
Before in LayerPanel component we had this [condition](https://github.com/elastic/kibana/pull/88578/files#diff-bd3389377551ea3146dd2faa0daf90bbbeb07a9fb55c86c9a2382c8473415d14L97):
```
if (	
    !datasourcePublicAPI ||	
    !props.activeVisualizationId ||	
    !props.visualizationMap[props.activeVisualizationId]	
  ) {	
    return null;	
  }
```
I needed to remove it from there as it made it very difficult to memoize `onDrop` function. Now after the changes, I am passing `activeVisualization` to LayerPanel so `!props.activeVisualizationId || !props.visualizationMap[props.activeVisualizationId]` should be ok to remove but I still need to check if `datasourcePublicAPI` exists so I am doing it in `ConfigPanel`.

Actually, I was not able to break this thing in production so I am not sure what use case it is, but there's a [test](https://github.com/elastic/kibana/pull/88578/files#diff-6c57a2d5477a62898fb785025516e582438fbe870c7fafc05d424ce6f9fd0d99R102) that broke when I modified LayerPanel. I'm going to move this test to ConfigPanel, but if you know more about this use case let me know! @wylieconlon perhaps you remember?

Reply:
Hm, it might prevent a crash if the visualization is talking about a layer which isn't known to the datasource. I think it's fine to leave it in for now.